### PR TITLE
SQL-2403: Obtain path for mongosqltranslate lib from DriverSettings

### PR DESCRIPTION
**Pull request by bucaojit**
Removes the hard-coded values and uses the `driver` property from DriverSettings to get the directory of the ODBC driver. 

Tested on Windows and Power BI with `mongosqltranslate.dll` in the `C:\Program Files\MongoDB\Atlas SQL ODBC Driver\bin\` directory.  It is able to find the library, it hits an error right after though, it's possible it's addressed in SQL-2288.  

```
2024-10-23 20:59:42 - INFO: Loaded the mongosqltranslate library from: C:\Program Files\MongoDB\Atlas SQL ODBC Driver\bin\mongosqltranslate.dll
2024-10-23 20:59:42 - ERROR: [Env_0x264ea5c5750][Conn_0x264ea5c59d0] SQLDriverConnectW:: [MongoDB][Core] Deserializing libmongosqltranslate response Document to CommandResponse failed with error: unknown variant `version`, expected one of `Translate`, `GetNamespaces`, `GetMongosqlTranslateVersion`, `CheckDriverVersion`, `Error`
```

### CODE REVIEW

**Comment by EthanHardyMongo** [comment]
Yeah, that error is fixed SQL-2288

**Review by nbagnard — APPROVED** [review]
LGTM! 👍

File: core/src/err.rs [hunk 1]
```diff
@@ -88,6 +88,9 @@ pub enum Error {
         "Multiple Documents were returned when getting the schema; however, only one was expected."
     )]
     MultipleSchemaDocumentsReturned(usize),
+
+    #[error("Library path error: {0}")]
```
Comment (EthanHardyMongo) [inline core/src/err.rs:92]:
```suggestion
    #[error("Library path error: {0}")]
```
[nit]

File: core/src/mongosqltranslate.rs [hunk 2]
```diff
@@ -42,36 +31,60 @@ fn get_library_name(library_type: &str) -> String {
...
+    let path = PathBuf::from(exe_dir).join(lib_name);
+    Ok(path)
 }
 
-// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library
+// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library.
+// The library is expected to be in the same directory as the MongoDB ODBC driver.
 // and is responsible for determining the library name and path.
 // The library name and path are determined based on the operating system and architecture.
 // It is stored in a static variable to ensure that it is only loaded once.
```
Comment (EthanHardyMongo) [inline core/src/mongosqltranslate.rs:76]:
```suggestion
// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library
// and is responsible for determining the library name and path.
// The library name and path are determined based on the operating system and architecture.
// Additionally, the library is expected to be in the same directory as the MongoDB ODBC driver.
// The library is stored in a static variable to ensure that it is only loaded once.
```
I think this makes it a little more clear.

**Review by EthanHardyMongo — APPROVED** [review]
LGTM. I just have two small suggestions. Also, I noticed the PR is set to merge into master, but I think it's supposed to go into the EAP branch. Unfortunately, you will probably have merge conflicts with my recent PR if you merge into the EAP branch. 

Additionally, the way I run the integration tests is based on the hard-coded path, so those might start failing as well.


For reference, this is how I load the library and run the tests in evergreen:
```
sudo mkdir -p /opt/mongodb/atlas-sql-odbc-driver/
sudo mv -f libmongosqltranslate.so /opt/mongodb/atlas-sql-odbc-driver/libmongosqltranslate.so
cargo test mongosqltranslate_tests -- --nocapture   
```

**Comment by bucaojit** [comment]
I was having trouble rebasing this PR after I changed the branches to merge into so I'll open a new PR.

**Pull request closed without merging**